spec/controllers/users_controller_spec.rb

```diff
@@ -1232,6 +1232,7 @@ def post_user
           guardian = Guardian.new(user)
           guardian.stubs(:ensure_can_edit!).with(user).raises(Discourse::InvalidAccess.new)
           Guardian.stubs(new: guardian).with(user)
+          Guardian.stubs(new: guardian).with(nil)
```

Comment:
`nil`, so on 404 page monthly top topics will be loaded assuming that the user is not logged in.